### PR TITLE
Add DataFrame.info

### PR DESCRIPTION
**Pull request by milesgranger**

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -1182,6 +1189,99 @@ def nunique(self, axis=0, dropna=True):
...
+        if buf is None:
+            import sys
+
+            buf = sys.stdout
+        lines = [str(type(self))]
+
+        if len(self.columns) == 0:
+            lines.append(f"{type(self.index._meta).__name__}: 0 entries")
+            lines.append(f"Empty {type(self).__name__}")
+            if PANDAS_GE_150:
```
Comment (phofl) [inline dask_expr/_collection.py:1209]: we don't support pandas < 2.0

File: dask_expr/_collection.py [hunk 2]
```diff
@@ -1182,6 +1189,99 @@ def nunique(self, axis=0, dropna=True):
...
+            return
+
+        # Group and execute the required computations
+        computations = {}
+        if verbose:
+            memory_usage = True
+            computations.update({"index": self.index, "count": self.count()})
+        if mem_use:
+            computations.update(
+                {"memory_usage": self.map_partitions(M.memory_usage, index=True)}
```
Comment (phofl) [inline dask_expr/_collection.py:1222]: can we call our own memory usage?

File: dask_expr/_collection.py [hunk 3]
```diff
@@ -1182,6 +1189,99 @@ def nunique(self, axis=0, dropna=True):
...
+            )
+        import dask.array as da
+
+        computations = dict(
+            zip(computations.keys(), da.compute(*computations.values()))
```
Comment (phofl) [inline dask_expr/_collection.py:1227]: Why do we need dask.array here?

File: dask_expr/tests/test_collection.py [hunk 4]
```diff
@@ -51,6 +53,66 @@ def test_del(pdf, df):
...
+    if buf and not verbose and not memory_usage:
+        expected = (
+            "<class 'dask_expr._collection.DataFrame'>\n"
+            "Columns: 2 entries, x to y\n"
+            "dtypes: int64(2)"
+        )
+        assert buf.getvalue() == expected
+    elif buf and verbose and not memory_usage:
+        expected = (
+            "<class 'dask_expr._collection.DataFrame'>\n"
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply dask_expr/tests/test_collection.py]: https://github.com/dask-contrib/dask-expr/pull/513/commits/da2f17142790e00693a7beecbf4fc0b6ff28bc08 gone!

File: dask_expr/_collection.py (hunk 3 above)
Comment (milesgranger) [reply dask_expr/_collection.py]: ya, naively just getting compute.. maybe this is better https://github.com/dask-contrib/dask-expr/pull/513/commits/a5f442bf103b59a63768811bf596680b35bb3976

File: dask_expr/_collection.py (hunk 2 above)
Comment (milesgranger) [reply dask_expr/_collection.py]: https://github.com/dask-contrib/dask-expr/pull/513/commits/a3fc272828987f680b6942baf0741d6326447818 missed that, sorry! :man_facepalming:

File: dask_expr/_collection.py (hunk 1 above)
Comment (milesgranger) [reply dask_expr/_collection.py]: https://github.com/dask-contrib/dask-expr/pull/513/commits/e918ff965fc46d90bc047cb68fa0ace22394ab87 done.

File: dask_expr/_collection.py (hunk 3 above)
Comment (phofl) [reply dask_expr/_collection.py]: Yep, thx

**Review by phofl — APPROVED** [review]

**Pull request merged**